### PR TITLE
[otelfiber] Allow for disabling client IP collection in traces

### PR DESCRIPTION
**Pull request by gzuidhof**
Introduces a `WithCollectClientIP(collect bool)` that disables IP collection into traces.

## Summary by CodeRabbit

- **New Features**
	- Enhanced privacy settings with the ability to control the collection of client IP addresses in requests.
	- Middleware function now collects client IP addresses by default.
	- Attribute handling now includes client IP based on configuration.
	- Improved middleware configuration using functional parameters.

### CODE REVIEW

File: otelfiber/semconv.go [hunk 1]
```diff
@@ -56,9 +56,11 @@ func httpServerTraceAttributesFromRequest(c *fiber.Ctx, cfg config) []attribute.
 	if username, ok := HasBasicAuth(c.Get(fiber.HeaderAuthorization)); ok {
 		attrs = append(attrs, semconv.EnduserIDKey.String(utils.CopyString(username)))
 	}
-	clientIP := c.IP()
-	if len(clientIP) > 0 {
-		attrs = append(attrs, semconv.HTTPClientIPKey.String(utils.CopyString(clientIP)))
+	if cfg.collectClientIP {
+		clientIP := c.IP()
```
Comment (gaby) [inline otelfiber/semconv.go:60]: Is c.IP() ever empty?

**Review by gaby — COMMENTED** [review]
Make sure to update the README with the new functionality

**Comment by gzuidhof** [comment]
I wasn't sure how to update the README.. The README currently describes the `config` fields, but they are private as functional parameters are used instead. This PR introduces a field which is entirely private, so it can not be accessed.

**Comment by gaby** [comment]
> I wasn't sure how to update the README.. The README currently describes the `config` fields, but they are private as functional parameters are used instead. This PR introduces a field which is entirely private, so it can not be accessed.

Fair!

**Review by gaby — APPROVED** [review]
LGTM

**Comment by ReneWerner87** [comment]
@gzuidhof Why is the config setting private and can only be changed using the help function? 

Regardless of the answer, please also include the help function with signature and explanation in the readme

Every feature without examples or explanation gives the user fewer hints and is therefore questioned, is not known or is not used 
Only transparency and hints help to avoid questions and make it usable for everyone

**Comment by gzuidhof** [comment]
I'm happy for you (or anybody else) to make changes to this PR as you see fit :) 

There is no mechanism for passing in your own `Config` object anyhow, so it doesn't really matter if it's public or private - so better to default to private was my thinking. Aside from that, with something that defaults to `true` setting it to false would not do anything.

The discoverability for these functional parameters are in your language server suggestions.. `otelfiber.With...`

Anyhow: happy for you to make any changes, just explaning my reasons :)

**Comment by ReneWerner87** [comment]
okay thanks for the explanation

can you include the signatures of the "with" methods and their explanation in the readme

**Comment by ReneWerner87** [comment]
f.e. https://docs.gofiber.io/api/middleware/proxy#signatures

**Comment by gzuidhof** [comment]
Done! :)

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
thx for the adjustment
@gzuidhof  can you also check these suggestions
image

from our ai

**Comment by gzuidhof** [comment]
I addressed them [in this commit](https://github.com/gofiber/contrib/pull/1033/commits/b5664dedb062cdf3e6fdaf335d93b846067d7a05), I think it hasn't refreshed

**Comment by ReneWerner87** [comment]
ok, thx

**Comment by ReneWerner87** [comment]
@coderabbitai can you pls check the last state again

**Pull request merged**